external/fv3fit/fv3fit/_shared/config.py

```diff
@@ -2,15 +2,128 @@
 import fsspec
 import yaml
 import os
-from typing import Optional, Union, Sequence, List
+from typing import Optional, Tuple, Union, Sequence, List, TypeVar, Type, Dict
+import xarray as xr
 
+from loaders import batches
+
+
+# T used to type hint factory method according to
+# https://github.com/python/typing/issues/58#issuecomment-326240794
+T = TypeVar("T", bound="Loadable")
 
 DELP = "pressure_thickness_of_atmospheric_layer"
 MODEL_CONFIG_FILENAME = "training_config.yml"
 
 
+KERAS_MODELS: Dict[str, type] = {}
+SKLEARN_MODEL_TYPES = ["sklearn", "rf", "random_forest", "sklearn_random_forest"]
+
+
+def get_keras_model(name):
+    return KERAS_MODELS[name]
+
+
+def register_keras_trainer(name):
+    def decorator(cls):
+        KERAS_MODELS[name] = cls
+        return cls
+
+    return decorator
+
+
+class Dumpable:
+    def dump(self, filename: str) -> None:
+        dict_ = dataclasses.asdict(self)
+        with fsspec.open(filename, "w") as f:
+            yaml.safe_dump(dict_, f)
+
+
+class Loadable:
+    @classmethod
+    def load(cls: Type[T], filename: str) -> T:
+        with fsspec.open(filename, "r") as f:
+            config_dict = yaml.safe_load(f)
+        return cls(**config_dict)  # type: ignore
+
+
+@dataclasses.dataclass
+class DataConfig(Dumpable):
+    """Convenience wrapper for model training data
+
+    Attrs:
+        variables: names of variables to include in dataset
+        batch_function: name of function from `fv3fit.batches` to use for
+            loading batched data
+        batch_kwargs: keyword arguments to pass to batch function
+        data_path: location of training data to be loaded by batch function
+        validation_timesteps: timestamps to use as validation samples
+        timesteps_source: one of "timesteps_file",
```

Comment:
`timesteps_source` and `validation_timesteps` deleted

Reply:
Removed.